src/components/application_manager/test/request_tracker_test.cc

```diff
@@ -0,0 +1,250 @@
+       TrackAppRequestInNone_NoLimits_ExpectAlwaysSuccess) {
+  const uint32_t app_id = 1;
+  const mobile_apis::HMILevel::eType none_level =
+      mobile_apis::HMILevel::HMI_NONE;
+
+  const uint32_t no_limit = 0;
+
+  EXPECT_CALL(mock_request_controller_settings_,
+              app_hmi_level_none_time_scale())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  EXPECT_CALL(mock_request_controller_settings_,
+              app_hmi_level_none_time_scale_max_requests())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  for (size_t i = 0; i < kDefaultAppHmiLevelNoneTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, none_level));
+  }
+
+  EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+            tracker_.Track(app_id, none_level));
+}
+
+TEST_F(RequestTrackerTestClass,
+       TrackAppRequestInOtherThanNone_ExpectSuccessTillLimit) {
+  const uint32_t app_id = 1;
+  mobile_apis::HMILevel::eType hmi_level =
+      mobile_apis::HMILevel::HMI_BACKGROUND;
+
+  SetDefaultConstraints();
+
+  for (size_t i = 0; i < kDefaultAppTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, hmi_level));
+    if (i % 2) {
+      hmi_level = mobile_apis::HMILevel::HMI_FULL;
+    } else {
+      hmi_level = mobile_apis::HMILevel::HMI_LIMITED;
+    }
+  }
+
+  EXPECT_EQ(application_manager::request_controller::TrackResult::
+                kMaxRequestsExceeded,
+            tracker_.Track(app_id, hmi_level));
+}
+
+TEST_F(RequestTrackerTestClass,
+       TrackAppRequestInOtherThanNone_NoLimits_ExpectAlwaysSuccess) {
+  const uint32_t app_id = 1;
+  mobile_apis::HMILevel::eType hmi_level =
+      mobile_apis::HMILevel::HMI_BACKGROUND;
+
+  const uint32_t no_limit = 0;
+
+  EXPECT_CALL(mock_request_controller_settings_, app_time_scale())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  EXPECT_CALL(mock_request_controller_settings_, app_time_scale_max_requests())
+      .WillRepeatedly(ReturnRef(no_limit));
+
+  for (size_t i = 0; i < kDefaultAppTimeScaleMaxRequests; ++i) {
+    EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+              tracker_.Track(app_id, hmi_level));
+    if (i % 2) {
+      hmi_level = mobile_apis::HMILevel::HMI_FULL;
+    } else {
+      hmi_level = mobile_apis::HMILevel::HMI_LIMITED;
+    }
+  }
+
+  EXPECT_EQ(application_manager::request_controller::TrackResult::kSuccess,
+            tracker_.Track(app_id, hmi_level));
+}
+
+TEST_F(RequestTrackerTestClass,
+       TrackTwoAppsRequestInNone_ExpectSuccessTillLimit) {
+  const uint32_t app_id_1 = 1;
+  const uint32_t app_id_2 = 2;
+
+  const mobile_apis::HMILevel::eType none_level =
+      mobile_apis::HMILevel::HMI_NONE;
+
+  SetDefaultConstraints();
+
+  for (size_t i = 0; i < kDefaultAppHmiLevelNoneTimeScaleMaxRequests; ++i) {
+    if (i % 2) {
```

Comment:
@LuxoftAKutsan 
This is simplest way I know to emulate requests coming from two applications one-by-one.